android/src/main/java/com/mapbox/mapboxgl/GlobalMethodHandler.java

```diff
@@ -61,6 +75,14 @@ private InputStream openTilesDbFile(String tilesDb) throws IOException {
         }
     }
 
+    private String extractAccessToken(MethodCall methodCall, String fallbackValue) {
+        if (methodCall.hasArgument("accessToken")) {
```

Comment:
@kleeb I'm getting a ClassCastException here every time I try to download an offline region. The reason seems to be that flutter's methods like methodCall.hasArgument() only work if you pass a flutter `Map` or `JSONObject` in flutter. Because you pass an already json-encoded String in global.dart:50 this fails. I got it to work by replacing all calls to extractAccessToken() with 
`gson.fromJson(methodCall.arguments.toString(), JsonObject.class).get("accessToken").getAsString()` 
or `new Gson().fromJson(methodCall.arguments.toString(), JsonObject.class).get("accessToken").getAsString()`

After that downloading definitely works. After I delete a region I can still view it offline but I guess that is just because of caching.

Reply:
Could you look into patching this @kleeb ?